### PR TITLE
Update of rl example: racing

### PR DESCRIPTION
**Pull request by Adaickalavan**
1. Updated to use `smarts[camera_obs]==0.7.0rc0`
2. Added action space
3. Updated loop scenario to use continuous flowing traffic
4. Avoided subprocess in scenario build
5. Simplified example folder structure

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -79,7 +79,7 @@ doc = sphinx>=4.4.0
     sphinxcontrib-apidoc>=0.3.0
 extras = pynput>=1.7.4  # Used by HumanKeyboardAgent
 gym = moviepy == 1.0.3
-remote_agent = grpcio==1.32.0
+remote-agent = grpcio==1.32.0
```
Comment (Gamenot) [inline setup.cfg:82]: Is this change necessary? The direction we are going is to have all of the extras remove dash.

**Review by Gamenot — APPROVED** [review]
This works, we will need to update the format of the main repository extras as a different PR.

File: setup.cfg (hunk 1 above)
Comment (Adaickalavan) [reply setup.cfg]: Reverted accidental change.

Comment (Gamenot) [reply setup.cfg]: On second review, this is the cause of all of the extras having the `-` -> `_` conversion.

Comment (Adaickalavan) [reply setup.cfg]: Shall we simply convert all extras to use `_`, to avoid confusion and to avoid auto name changes?

Comment (Gamenot) [reply setup.cfg]: I think we can do that because the next release is a major version change.

Comment (Gamenot) [reply setup.cfg]: For now we should use `remote-agent` and then have an issue to convert the rest.

Comment (Adaickalavan) [reply setup.cfg]: Conversion of `-` to `_` in `options.extras_require` is done in #1756 .

File: examples/rl/racing/setup.py [hunk 2]
```diff
@@ -15,10 +15,10 @@
     packages=find_packages(),
     include_package_data=True,
     zip_safe=True,
-    python_requires=">=3.7",
+    python_requires="~=3.7",
```
Comment (Gamenot) [inline examples/rl/racing/setup.py:18]: `3.8` does not work?

**Review by saulfield — APPROVED** [review]

File: examples/rl/racing/setup.py (hunk 2 above)
Comment (Gamenot) [reply examples/rl/racing/setup.py:18]: We can merge this
